content/en/network_monitoring/devices/troubleshooting.md

```diff
@@ -12,91 +12,165 @@ further_reading:
+   DROP       all  --  anywhere             10.4.5.6
+   ```
 
-Datadog uses SNMP to discover devices. During discovery, the SNMP port (default 161) is polled. If there's a response and a profile to match, this is considered a discovered device.
+   - Ensure your community string matches.
 
-#### Does Datadog do MIB certification? Do I need to send you all my MIBs? How do I convert my MIBs with Python?
+   **Incorrect SNMPv2 credentials**:
 
-The Datadog Agent is MIB-less, meaning you don't need to do anything with your MIBs. All metrics collected with Datadog device profiles automatically work without the MIB.
+   ``` 
+   Error: an authentication method needs to be provided
+   ```
 
-To add metrics or a custom configuration, list the MIB name, table name, table OID, symbol, and symbol OID, for example:
+   **Solution**: 
+   If you are using SNMPv2 you need to set a community string.
 
-```yaml
-- MIB: EXAMPLE-MIB
-    table:
-      # Identification of the table which metrics come from.
-      OID: 1.3.6.1.4.1.10
-      name: exampleTable
-    symbols:
-      # List of symbols ('columns') to retrieve.
-      # Same format as for a single OID.
-      # Each row in the table emits these metrics.
-      - OID: 1.3.6.1.4.1.10.1.1
-        name: exampleColumn1
-```
+   **Incorrect SNMPv3 privacy protocol**
 
-#### Can I still use Network Device Monitoring if my device-model pair isn't supported?
+   ```
+   Error: check device reachable: failed: decryption error; failed to autodetect profile: failed to fetch sysobjectid: cannot get sysobjectid: decryption error; failed to fetch values: failed to fetch scalar oids with batching: failed to fetch scalar oids: fetch scalar: error getting oids `[1.3.6.1.2.1.1.1.0 1.3.6.1.2.1.1.2.0 1.3.6.1.2.1.1.3.0 1.3.6.1.2.1.1.5.0]`: decryption error  OR Error: check device reachable: failed: wrong digest; failed to autodetect profile: failed to fetch sysobjectid: cannot get sysobjectid: wrong digest; failed to fetch values: failed to fetch scalar oids with batching: failed to fetch scalar oids: fetch scalar: error getting oids `[1.3.6.1.2.1.1.1.0 1.3.6.1.2.1.1.2.0 1.3.6.1.2.1.1.3.0 1.3.6.1.2.1.1.5.0]`: wrong digest
+   ```
 
-Datadog collects generic base-line metrics from all devices. If there are unsupported metrics from a vendor MIB, you can write a custom profile, or send a feature request to [Datadog support][1].
+   **Solution**: 
+   Ensure the following configuration parameters are correct: `user`, `authKey`, `authProtocol`, `privKey`, `privProtocol`.
 
-If you send a feature request, Datadog support needs a `snmpwalk` from the requested device. Run the following and send the output:
+### Traps not being received for devices
 
-```
-snmpwalk -O bentU -v 2c -c <COMMUNITY_STRING> <IP_ADDRESS>:<PORT> 1.3.6
-```
+1. Check the Datadog `agent.log` file to ensure that you aren't unable to bind to the traps port. If this is the case, the following error is present:
```

Comment:
```suggestion
Check the Datadog `agent.log` file to ensure that you can bind to the traps port. The following error indicates that you are unable to bind to the traps port:
```
this shouldn't be in the ordered list—the ordered list should just be the "solution" steps
also clearing up some unnecessarily convoluted language